openpype/hosts/maya/plugins/publish/extract_maya_scene_raw.py

```diff
@@ -1,13 +1,118 @@
 # -*- coding: utf-8 -*-
 """Extract data as Maya scene (raw)."""
 import os
+import contextlib
 
 from maya import cmds
 
 from openpype.hosts.maya.api.lib import maintained_selection
 from openpype.pipeline import AVALON_CONTAINER_ID, publish
 
 
+def offset_node(node, offset):
+    if cmds.nodeType(node).startswith("animCurve"):
+        cmds.keyframe(node, edit=True, relative=True, timeChange=offset)
+
+    if cmds.nodeType(node) == "imagePlane":
+        node_attr = node + ".frameOffset"
+        cmds.setAttr(node_attr, cmds.getAttr(node_attr) + (-offset))
+
+    if cmds.nodeType(node) == "timeSliderBookmark":
+        for attr in ["timeRangeStart", "timeRangeStop"]:
+            node_attr = "{}.{}".format(node, attr)
+            cmds.setAttr(node_attr, cmds.getAttr(node_attr) + offset)
+
+
+@contextlib.contextmanager
+def offset_scene(offset):
+    # Exit early for easier stacking of context managers.
+    if offset is None:
+        yield
+        return
+
+    nodes = cmds.ls(type="animCurve")
+    nodes += cmds.ls(type="imagePlane")
+    nodes += cmds.ls(type="timeSliderBookmark")
+    changed_nodes = []
+    try:
+        for node in nodes:
+            offset_node(node, offset)
```

Comment:
You might need to exclude referenced nodes since you can't as far as I know offset referenced animtation curves unless that option is specifically enabled.